src/pages/status/index.tsx

```diff
@@ -1,48 +1,170 @@
+import ChequebookDeployFund from './SetupSteps/ChequebookDeployFund'
+import PeerConnection from './SetupSteps/PeerConnection'
+import { Context } from '../../providers/Bee'
 
 const useStyles = makeStyles((theme: Theme) =>
   createStyles({
     root: {
+      padding: theme.spacing(2),
       width: '100%',
-      display: 'grid',
-      rowGap: theme.spacing(3),
+    },
+    button: {
+      marginTop: theme.spacing(1),
+      marginRight: theme.spacing(1),
+    },
+    actionsContainer: {
+      margin: theme.spacing(2),
     },
   }),
 )
 
-export default function Status(): ReactElement {
+interface Step {
+  label: string
+  isOk: boolean
+  component: ReactElement
+}
+
+export default function NodeSetupWorkflow(): ReactElement {
   const classes = useStyles()
+  const [activeStep, setActiveStep] = useState(-1)
 
   const {
     status,
+    isLoading,
     latestUserVersion,
+    latestPublishedVersion,
     isLatestBeeVersion,
     latestBeeVersionUrl,
     topology,
     nodeAddresses,
     chequebookAddress,
-  } = useContext(BeeContext)
+  } = useContext(Context)
+
+  const steps: Step[] = [
+    {
+      label: 'Connected to Node DebugAPI',
+      isOk: status.debugApiConnection,
+      component: <DebugConnectionCheck isOk={status.debugApiConnection} />,
+    },
+    {
+      label: 'Running latest Bee version',
+      isOk: status.version,
+      component: (
+        <VersionCheck
+          isOk={status.version}
+          isLatestBeeVersion={isLatestBeeVersion}
+          userVersion={latestUserVersion}
+          latestVersion={latestPublishedVersion}
+          latestUrl={latestBeeVersionUrl}
+        />
+      ),
+    },
+    {
+      label: 'Connected to xDai Blockchain',
+      isOk: status.blockchainConnection,
+      component: <EthereumConnectionCheck isOk={status.blockchainConnection} nodeAddresses={nodeAddresses} />,
+    },
+    {
+      label: 'Deployed and Funded Chequebook',
+      isOk: status.chequebook,
+      component: (
+        <ChequebookDeployFund chequebookAddress={chequebookAddress?.chequebookAddress} isOk={status.chequebook} />
+      ),
+    },
+    {
+      label: 'Connected to Node API',
+      isOk: status.apiConnection,
+      component: <NodeConnectionCheck isOk={status.apiConnection} />,
+    },
+    {
+      label: 'Connected to Peers',
+      isOk: status.topology,
+      component: <PeerConnection isOk={status.topology} topology={topology} />,
+    },
+  ]
+
+  useEffect(() => {
+    // If the user already changed the active step we don't want to overwrite it
+    if (activeStep >= 0 && activeStep < steps.length) return
+
+    // If any step is not fully loaded yet return
+    if (!isLoading) return
+
+    // Select first step that is not OK
+    // This is deliberately a for loop (and not forEach) so that we can terminate the useEffect from within the cycle
+    for (let i = 0; i < steps.length; ++i) {
+      if (!steps[i].isOk) {
+        setActiveStep(i)
+
+        return
+      }
```

Comment:
Setting state in a loop - does it have any performance tax? Or is it irrelevant because it is in `useEffect`?